packages/nextjs/src/utils/instrumentServer.ts

```diff
@@ -228,3 +285,16 @@ function shouldTraceRequest(url: string, publicDirFiles: Set<string>): boolean {
   // `static` is a deprecated but still-functional location for static resources
   return !url.startsWith('/_next/') && !url.startsWith('/static/') && !publicDirFiles.has(url);
 }
+
+function addRequestDataToEvent(event: SentryEvent, req: NextRequest): SentryEvent {
+  event.request = {
+    // TODO body/data
+    url: req.url.split('?')[0],
+    cookies: req.cookies,
+    headers: req.headers as { [key: string]: string },
```

Comment:
Why do we have to cast? Can’t we just change the NextRequest type?

Reply:
Good point.